src/csharp/Grpc.Tools.MSBuildIntegration/Grpc.Tools.MSBuildIntegration.targets

```diff
@@ -0,0 +1,27 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <CoreCompileDependsOn>Proto;$(CoreCompileDependsOn)</CoreCompileDependsOn>
+  </PropertyGroup>
+  <ItemGroup>
+    <AvailableItemName Include="Proto"/>
+  </ItemGroup>
+  <Target Name="Proto" Inputs="@(Proto)" Outputs="@(Proto->'$(IntermediateOutputPath)%(FileName).cs')">

```

Comment:
> Comment 1: See lines 20 and 23, I transform the input Proto files into filenames with both variations.
I do not think I understand how it works. Suppose you have 
```
<Target Name="Proto" Inputs="@(Proto)" Outputs="@(Proto->'$(IntermediateOutputPath)%(FileName).cs')">
```
In my understanding, if your `Proto` collection has the only item `some_proto.proto`, then the Outputs will be  `obj/Release/some_proto.cs`. Since that file will not be made by `protoc` ever, the target will be rerun every time. The correct name is `SomeProto.cs`.